src/main/java/com/iota/iri/network/pipeline/SolidifyStage.java

```diff
@@ -0,0 +1,94 @@
+package com.iota.iri.network.pipeline;
+
+import com.iota.iri.controllers.TipsViewModel;
+import com.iota.iri.controllers.TransactionViewModel;
+import com.iota.iri.model.Hash;
+import com.iota.iri.service.validation.TransactionSolidifier;
+import com.iota.iri.storage.Tangle;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.iota.iri.controllers.TransactionViewModel.fromHash;
+
+/**
+ * The {@link SolidifyStage} is used to process newly received transaction for solidity. Once a transaction has been
+ * passed from the {@link ReceivedStage} it will be placed into this stage to have the {@link TransactionSolidifier}
+ * check the solidity of the transaction. If the transaction is found to be solid, it will be passed forward to the
+ * {@link BroadcastStage}. If it is found to be unsolid, it is put through the solidity check so that missing reference
+ * transactions get requested. If the transaction is unsolid, a random solid tip is broadcast instead to keep the
+ * requests transmitting to neighbors.
+ */
+public class SolidifyStage implements Stage {
+    private static final Logger log = LoggerFactory.getLogger(SolidifyStage.class);
+
+    private TransactionSolidifier txSolidifier;
+    private TipsViewModel tipsViewModel;
+    private Tangle tangle;
+
+    /**
+     * Constructor for the {@link SolidifyStage}.
+     *
+     * @param txSolidifier       Transaction validator implementation for determining the validity of a transaction
+     * @param tipsViewModel     Used for broadcasting random solid tips if the subject transaction is unsolid
+     * @param tangle            A reference to the nodes DB
+     */
+    public SolidifyStage(TransactionSolidifier txSolidifier, TipsViewModel tipsViewModel, Tangle tangle){
+        this.txSolidifier = txSolidifier;
+        this.tipsViewModel = tipsViewModel;
+        this.tangle = tangle;
+    }
+
+    /**
+     * Processes the payload of the {@link ProcessingContext} as a {@link SolidifyPayload}. First the transaction will
+     * be checked for solidity and validity. If the transaction is already solid or can be set solid quickly by the
+     * transaction validator, the transaction is passed to the {@link BroadcastStage}. If not, a random solid tip is
```

Comment:
can be set solid quickly by the transaction ~validator~ solidifier

Reply:
Aaah nice catch thought I caught them all. Thanks :+1: